blackduck/ClientCore.py

```diff
@@ -78,11 +82,12 @@ def _get_items(self, url, method='GET', page_size=100, name='', **kwargs):
         page_size (int, optional): [description]. Defaults to 100.
         name (str, optional): [description]. Defaults to ''.
 
-    Yields:
-        [type]: [description]
+    Returns:
+        list(dict/json): list of items
```

Comment:
I had quite a bit of difficulty using the generator when I wanted to pick out specific elements, check length, etc. I found I was always casting the generator to a list first.

I don't think the generator buys anything over a list unless we implement next() and only fetch subsequent pages when needed.

Reply:
Well - the generator does only fetch subsequent pages when needed, when used in a for loop.
if you set the requests log level to debug, you can see it working in this sample:
```
counter = 1
for project in bd.get_resource(name='projects'):
    input(f"{counter}: {project.get('name')}")
    counter+=1
```

I specifically chose to use a generator here as it allowed me to operate on resource items in bite sized chunks - which is essential when running in a restricted environment (AWS Lambda in my case). We have a lambda that runs nightly to check the status of projects/versions/vulnerable-components & update our dashboards. Just to give you an idea - we have ~750 projects. 

I wrote get_metadata mainly for access to `totalCount`, which could be used instead of `len(resource)`.

Reply:
Nod, understood. I'll revert this method back to being a generator before merging into master.